cdap-proto/src/main/java/co/cask/cdap/proto/StreamProperties.java

```diff
@@ -28,21 +28,29 @@
 
   private final Long ttl;
   private final FormatSpecification format;
+  @SerializedName("owner.principal")
+  private final String ownerPrincipal;
 
   @SerializedName("notification.threshold.mb")
   private final Integer notificationThresholdMB;
   private final String description;
 
   public StreamProperties(Long ttl, FormatSpecification format, Integer notificationThresholdMB) {
-    this(ttl, format, notificationThresholdMB, null);
+    this(ttl, format, notificationThresholdMB, null, null);
   }
 
   public StreamProperties(Long ttl, FormatSpecification format, Integer notificationThresholdMB,
                           @Nullable String description) {
+    this(ttl, format, notificationThresholdMB, description, null);
+  }
+
+  public StreamProperties(Long ttl, FormatSpecification format, Integer notificationThresholdMB,
+                          @Nullable String description, @Nullable String ownerPrincipal) {
```

Comment:
When we represents owner principal as `String` vs `KerberosPrincipalId`? Shouldn't we always use the `KerberosPrincipalId` class?